### PR TITLE
Add Tools menu and slit calculator to reflectometry GUI

### PR DESCRIPTION
**Pull request by gemmaguest**
**Description of work.**
This PR adds a Tools menu to the ISIS Reflectometry interface, containing an option to open the Slit Calculator dialog (which is an existing dialog).

Note that there are deliberately some TODOs left in the code because outline functions have also been added for the Options dialog but this will be implemented in a future PR as the remainder of issue #26533

**To test:**

- Open the ISIS Reflectometry interface
- On the Tools menu, select Slit Calculator. The Slit Calculator dialog should open showing the default values in the first 2 boxes for the selected instrument (see screenshots below for correct values).
- Check that you can edit the input values in the Slit Calculator.
- Change the instrument on the ISIS Reflectometry interface and check that the default values on the Slit Calculator change appropriately.
- Close the Slit Calculator, change the instrument, and re-open it. Again, check that the default values are correct.
- Edit the values for resolution, footprint and angle. These should not change when you change the instrument or close and re-open the slit calculator. They do however get reset if you close and reopen the reflectometry GUI.

INTER
inter

SURF
surf

CRISP
crisp

POLREF
polref

OFFSPEC
offspec

Refs #26533 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####
 and that the output values are updated
- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp [hunk 1]
```diff
@@ -36,15 +38,17 @@ Mantid::Kernel::Logger g_log("Reflectometry GUI");
...
  */
 MainWindowPresenter::MainWindowPresenter(
     IMainWindowView *view, IMessageHandler *messageHandler,
+    std::unique_ptr<ISlitCalculator> slitCalculator,
     std::unique_ptr<IBatchPresenterFactory> batchPresenterFactory)
-    : m_view(view), m_messageHandler(messageHandler),
-      m_batchPresenterFactory(std::move(batchPresenterFactory)),
-      m_instrument() {
+    : m_view(view), m_messageHandler(messageHandler), m_instrument(),
+      m_slitCalculator(std::move(slitCalculator)),
```
Comment (giovannidisiena) [inline qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: Is it safe to use `std::move` with `std::unique_ptr`?

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/IMainWindowView.h [hunk 2]
```diff
@@ -29,6 +29,8 @@ class MainWindowSubscriber {
   virtual void notifyCloseBatchRequested(int) = 0;
   virtual void notifySaveBatchRequested(int) = 0;
```
Comment (giovannidisiena) [inline qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/IMainWindowView.h]: Can the `int` arguments be removed from these pure virtual functions?

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp (hunk 1 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: Yes, this transfers ownership to the new unique_ptr and resets the old one

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/IMainWindowView.h (hunk 2 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/IMainWindowView.h]: I don't think so - it looks like that's the batch index which I assume is used in the concrete implementation, and the signature here needs to match (although I haven't changed that in this PR so not sure how it's used off the top of my head).

Comment (giovannidisiena) [reply qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/IMainWindowView.h]: Yes, that makes sense!

**Comment by gemmaguest** [comment]
@giovannidisiena The behaviour you saw with resolution, footprint and angle not changing for different instruments is the same in v3.13.0 so I believe this is correct. Similarly with those values only being reset when you close the reflectometry GUI. I've updated the test instructions accordingly.

**Comment by giovannidisiena** [comment]
> @giovannidisiena The behaviour you saw with resolution, footprint and angle not changing for different instruments is the same in v3.13.0 so I believe this is correct. Similarly with those values only being reset when you close the reflectometry GUI. I've updated the test instructions accordingly.

In which case the functional testing passes 👍

**Review by giovannidisiena — APPROVED** [review]
Code looks good, let's :shipit:

**Pull request merged**